newt/sysinit/sysinit.go

```diff
@@ -181,6 +180,10 @@ func ResolveStageFuncsOrder(sfs []stage.StageFunc) ([]stage.StageFunc, error) {
 	// direct dependencies between each node of stage X to each node of
 	// stage Y to make sure they can be resolved properly and reordered
 	// if needed due to other dependencies.
+	if stages == nil {
+	  util.FmtNewtError("Cannot resolve sysinit order: no stages defined.")
+		return []stage.StageFunc{}, nil
+	}
```

Comment:
I think this can only happen if `sfs` is empty so you can basically check that at the very beginning of the function. Also no need to emit warning.